packages/dcos-integration-test/extra/resiliency/test_aws_cf.py

```diff
@@ -58,13 +58,15 @@ def get_running_agents():
     cluster.post('/mesos/machine/down', json=down_hosts).raise_for_status()
 
     # Wait for replacements
-    test_util.helpers.wait_for_len(get_running_agents, len(cluster.all_slaves), 600)
+    test_util.helpers.wait_for_len(partial(get_running_agents, 'SlaveServerGroup'), len(cluster.slaves), 600)
+    test_util.helpers.wait_for_len(
+        partial(get_running_agents, 'PublicSlaveServerGroup'), len(cluster.public_slaves), 600)
 
     # Reset the cluster to have the replacement agents
-    cluster.slaves = sorted([agent.private_ip for agent in
-                             dcos_launchpad.get_private_agent_ips(state_list=['running'])])
-    cluster.public_slaves = sorted([agent.private_ip for agent in
-                                    dcos_launchpad.get_public_agent_ips(state_list=['running'])])
+    cluster.slaves = sorted([agent.private_ip_address for agent in
```

Comment:
nit / for future: Would be nice if we could somehow just tell the ClusterApi object that "here's the new set of slaves" and it would deal with the sorted() bit. It's odd that this has to magically know they're sorted, etc. Would be nice to keep all that normalization and list validation (no repeats, no masters, etc) in one place